reactor-core/src/test/java/reactor/util/ConsoleLoggerTest.java

```diff
@@ -296,4 +305,67 @@ public void formatNull() {
 				.isEqualTo("[ INFO] (" + Thread.currentThread().getName() + ") null\n");
 	}
 
+	/**
+	 * Ensure console logger factory synchronizes logger acquisition properly.
+	 */
+	@Test
+	public void getConsoleLoggerShouldBeThreadSafe() throws Exception {
```

Comment:
there is a `RaceTestUtils` utility class in the test sourceset that tries to coordinate runnables so that they run at the same time (trigger a race condition), in a more lightweight way than with eg. a jcstress test. maybe this could apply and simplify here?

Reply:
Just what I needed, thanks. I've reworked the test to use it.